Nightly Cuda build failure: cuda/12.6.1+gcc/11.4.0 Volta arch, "Kokkos_DualView.hpp(473): error: static assertion failed..."

**Describe the bug**

Nightly builds with cuda/12.6.1+gcc/11.4.0 Volta arch, and Cuda Serial and OpenMP backends enabled, fail to compile with static_assert:

```
17:46:43 /home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_cuda1261_gcc1140_cpp17/kokkos/containers/unit_tests/../src/Kokkos_DualView.hpp(473): error: static assertion failed with "Template parameter to .sync() must exactly match one of the DualView's device types or one of the execution or memory spaces"
17:46:43       static_assert(
17:46:43       ^
17:46:43           detected during:
17:46:43             instantiation of "int Kokkos::DualView<DataType, Properties...>::get_device_side<Device>() [with DataType=double *, Properties=<Kokkos::LayoutLeft, Test::<unnamed>::DeviceType>, Device=Kokkos::Serial]" at line 796
17:46:43             instantiation of "void Kokkos::DualView<DataType, Properties...>::modify<Device>() [with DataType=double *, Properties=<Kokkos::LayoutLeft, Test::<unnamed>::DeviceType>, Device=Kokkos::Serial]" at line 763 of /home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_cuda1261_gcc1140_cpp17/kokkos/containers/unit_tests/TestDualView.hpp
17:46:43 
17:46:43 1 error detected in the compilation of "/home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_cuda1261_gcc1140_cpp17/Testing_2025-04-10_17.33.03/containers/unit_tests/serial/TestSerial_DualView.cpp".
17:46:44 make[2]: *** [containers/unit_tests/CMakeFiles/Kokkos_ContainersUnitTest_Serial.dir/build.make:107: containers/unit_tests/CMakeFiles/Kokkos_ContainersUnitTest_Serial.dir/serial/TestSerial_DualView.cpp.o] Error 1
17:46:44 make[2]: *** Waiting for unfinished jobs....
17:46:44 make  -f containers/performance_tests/CMakeFiles/Kokkos_ContainersPerformanceTest_Cuda.dir/build.make containers/performance_tests/CMakeFiles/Kokkos_ContainersPerformanceTest_Cuda.dir/depend
```

This follows after merge of these commits

**Changes**

    DualView: Don't allocate modified_flags in modify ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda1261_gcc1140_cpp17/659/changes#bd8bcc8893dec9560a8d569271847dfac045e66f) / [githubweb](https://github.com/kokkos/kokkos/commit/bd8bcc8893dec9560a8d569271847dfac045e66f))
    Vector: modify after resize ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda1261_gcc1140_cpp17/659/changes#5b8aff2c060451c2be2980cd2a0bd66373988e4f) / [githubweb](https://github.com/kokkos/kokkos/commit/5b8aff2c060451c2be2980cd2a0bd66373988e4f))
    Add test checking behavior for default constructed DualViews ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda1261_gcc1140_cpp17/659/changes#49a7bd11b9114e9cdf67bdb2c983f650c511db80) / [githubweb](https://github.com/kokkos/kokkos/commit/49a7bd11b9114e9cdf67bdb2c983f650c511db80))
    DualView: Constructor with two Views make unique allocation check more strict and correct ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda1261_gcc1140_cpp17/659/changes#b6c1e875d738124acbfcb09908787fa7fda9bc61) / [githubweb](https://github.com/kokkos/kokkos/commit/b6c1e875d738124acbfcb09908787fa7fda9bc61))
    Enable bugprone-unhandled-exception-at-new check (#7968) ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda1261_gcc1140_cpp17/659/changes#02e37ba5537117001b1a4716c9e483483cf6e226) / [githubweb](https://github.com/kokkos/kokkos/commit/02e37ba5537117001b1a4716c9e483483cf6e226))
    Enable bugprone-suspicious-memory-comparison check (#7980) ([details](https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_cuda1261_gcc1140_cpp17/659/changes#47b04a637c6b661210f880c47198377703226b79) / [githubweb](https://github.com/kokkos/kokkos/commit/47b04a637c6b661210f880c47198377703226b79))

**Please include the following for a minimal reproducer**

**Reproducer notes**:
SHA: 47b04a637c6b661210f880c47198377703226b79

```bash

# Env (modules on kokkos-dev-2)
module load cmake/3.30.2 cuda/12.6.1 gcc/11.4.0

# Configure
cmake -DCMAKE_CXX_COMPILER=$KOKKOS_PATH/bin/nvcc_wrapper -DCMAKE_CXX_FLAGS="-Wall -Werror -Wno-deprecated-declarations" -DKokkos_ENABLE_OPENMP=ON -DKokkos_ENABLE_SERIAL=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_VOLTA70=ON -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_EXAMPLES=OFF -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_BUILD_TYPE=RELEASE -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF -DKokkos_ENABLE_DEPRECATED_CODE_4=OFF $KOKKOS_PATH

# Build
make -j
```

--- a/containers/unit_tests/TestDualView.hpp
+++ b/containers/unit_tests/TestDualView.hpp
@@ -760,10 +760,10 @@ TEST(TEST_CATEGORY, dualview_sequential_host_init) {
 TEST(TEST_CATEGORY, dualview_default_constructed) {
   DualViewType dv;
 
-  dv.modify<TEST_EXECSPACE>();
+  dv.modify<DualViewType::t_dev>();
   ASSERT_FALSE(dv.need_sync_host());
   ASSERT_FALSE(dv.need_sync_device());
-  dv.sync<TEST_EXECSPACE>();
+  dv.sync<DualViewType::t_dev>();
 
   dv.modify_host();
   ASSERT_FALSE(dv.need_sync_host());

